Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            var activity = _context.Activity.Where(a => a.ID == activityID)
+                            .Select(a => new ActivityViewModel
+                            {
+                                ID = a.ID,
+                                Name = a.Name,
+                                RegistrationStart = a.RegistrationStart,
+                                RegistrationEnd = a.RegistrationEnd,
+                                Sport = _context.Sport
+                                        .FirstOrDefault(s => s.ID == a.SportID),
+                                Status = _context.ActivityStatus
+                                        .FirstOrDefault(s => s.ID == a.StatusID)
+                                        .Description,
+                                MinCapacity = a.MinCapacity,
+                                MaxCapacity = a.MaxCapacity,
+                                SoloRegistration = a.SoloRegistration,
+                                Logo = a.Logo,
+                                Completed = a.Completed,
+                                Series = _seriesService.GetSeriesByActivityID(a.ID),
+                                Team = a.Team.Select(t => new TeamViewModel
+                                {
+                                    ID = t.ID,
+                                    Name = t.Name,
+                                    Status = _context.TeamStatus
+                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                                .Description,
+                                                
+                                    Logo = t.Logo
+                                })
+                            
+                            })
+                            .FirstOrDefault();
+            return activity;
+        }
+        public async Task UpdateActivity(UpdateActivityViewModel updatedActivity)
+        {
+            var activity = await _context.Activity.FindAsync(updatedActivity.ID);
+            activity.Name = updatedActivity.Name ?? activity.Name;
+            activity.Logo = updatedActivity.Logo ?? activity.Logo;
+            activity.RegistrationStart = updatedActivity.RegistrationStart ?? activity.RegistrationStart;
+            activity.RegistrationEnd = updatedActivity.RegistrationEnd ?? activity.RegistrationEnd;
+            activity.SportID = updatedActivity.SportID ?? activity.SportID;
+            activity.StatusID = updatedActivity.StatusID ?? activity.StatusID;
+            activity.MinCapacity = updatedActivity.MinCapacity ?? activity.MinCapacity;
+            activity.MaxCapacity = updatedActivity.MaxCapacity ?? activity.MaxCapacity;
+            activity.MaxCapacity = updatedActivity.MaxCapacity ?? activity.MaxCapacity;
+            activity.SoloRegistration = updatedActivity.SoloRegistration ?? activity.SoloRegistration;
+            activity.Completed = updatedActivity.Completed ?? activity.Completed;
```

Comment:
Your puts routes should technically replace every field of the object no matter if it was specified in the upload view model or not. All of these null coalescers bypass that.

Reply:
My goal with this was so that the UI could pass a single object like:
{
ID: 12345567
Name: "updated name"
Status: 3
}

Thus they wouldn't have to pass unnecessarily larger objects if there were only a couple updates. Perhaps, it would be more accurate to rename my "PUT" routes into "PATCH" routes? @EjPlatzer what do you think about that?

Reply:
So what you described there is a PATCH operation, not a PUT operation. I would say it's best to go and change those PUT routes to PATCHes. Just make sure you keep them as PUTs where it is actually a replacement.